### PR TITLE
Extend-product-service-with-single-product-getter_CU-jfb8wq

### PR DESCRIPTION
**Pull request by a-wilkes**

### CODE REVIEW

**Comment by a-wilkes** [comment]
Task linked: [CU-jfb8wq Extend product service with single product getter](https://app.clickup.com/t/jfb8wq)

File: src/app/shared/models/classes/product.interface.ts [hunk 1]
```diff
@@ -6,4 +6,5 @@ export interface Product {
     price: string;
     description: string;
     type: ProductType;
+    slug: string;
```
Comment (lpboucher) [inline src/app/shared/models/classes/product.interface.ts]: nice, this should indeed be a property of product

File: src/app/core/products/product.service.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    constructor() { }
+
+    ngOnInit() {
+    }
+
+    getProducts(): Observable<Coffee[]> {
+        return of(this.products);
+    }
+
+    getOne(slug: string): Observable<Coffee | undefined> {
```
Comment (lpboucher) [inline src/app/core/products/product.service.ts]: Typescript complaining about result of find being undefined right?

Going forward, we will have an endpoint to getasingle product and we will route to a 404 page if the slug does not exist. So we can leave as is for now

**Review by lpboucher — CHANGES_REQUESTED** [review]
Looks good, from my side.

Maybe you can rebase on the product-service after you have made the changes there

**Review by lpboucher — APPROVED** [review]

**Pull request merged**